core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -408,7 +408,8 @@ private[spark] class ExecutorAllocationManager(
       executorsPendingToRemove.add(executorId)
       true
     } else {
-      logWarning(s"Unable to reach the cluster manager to kill executor $executorId!")
+      logWarning(s"Unable to reach the cluster manager to kill executor $executorId!" +
+        s"Or no executor to kill!")
```

Comment:
Two things about this now:

- the message reads a little bit weirdly. How about: "Unable to send kill request for $executorId to cluster manager, or executor is busy."

- warning seems bit strong when the most probable cause of returning false is that the executor is busy; if there's a failure in sending a message to the cluster manager, I'm sure we'll see error / warning logs from other parts of the code.